app/src/main/java/co/smartreceipts/android/persistence/database/tables/AbstractColumnTable.java

```
@@ -91,28 +91,84 @@ public synchronized void onUpgrade(@NonNull SQLiteDatabase db, int oldVersion, i
+                    getTableName(), AbstractColumnTable.COLUMN_TYPE);
+            Logger.debug(this, addNewColumn);
+            db.execSQL(addNewColumn);
+
+            // finding new column types for old values (adding default column if correct type wasn't found)
+            Cursor columnsCursor = null;
+            try {
+                columnsCursor = db.query(getTableName(), new String[]{idColumnName, typeColumnName, COLUMN_TYPE}, null, null, null, null, null);
+                if (columnsCursor != null && columnsCursor.moveToFirst()) {
+                    final int idIndex = columnsCursor.getColumnIndex(idColumnName);
+                    final int typeIndex = columnsCursor.getColumnIndex(typeColumnName);
+
+                    do {
+                        final int id = columnsCursor.getInt(idIndex);
+                        final String oldColumnType = columnsCursor.getString(typeIndex);
+
+                        int newColumnType = receiptColumnDefinitions.getDefaultInsertColumn().getType();
+
+                        if (receiptColumnDefinitions instanceof ColumnFinder) {
+                            final int columnTypeByHeaderValue = ((ColumnFinder) receiptColumnDefinitions).getColumnTypeByHeaderValue(oldColumnType);
+                            if (columnTypeByHeaderValue >= 0) {
+                                newColumnType = columnTypeByHeaderValue;
+                            }
+                        }
+
+                        final ContentValues columnValues = new ContentValues(1);
+                        columnValues.put(COLUMN_TYPE, newColumnType);
+                        Logger.debug(this, "Updating old column header value: {} to new column type {}", oldColumnType, newColumnType);
+
+                        if (db.update(getTableName(), columnValues, COLUMN_ID + "= ?", new String[]{Integer.toString(id)}) == 0) {
+                            Logger.error(this, "Column update error happened");
```

Let's throw an exception here, since I'd prefer that we crash the app (to properly resolve this) rather than silently masking the error and potentially causing the user to lose his/her columns